### ISSUE TITLE
Add shortcodes

### ISSUE THREAD

**Issue by Keats** [getzola/zola#9]
Or an equivalent. It could hook into the markdown renderer and render a template when encountering one

**Comment by Keats** [comment]
Wordpress syntax is nicer than Hugo's: https://en.support.wordpress.com/shortcodes/ imo but that would conflict with Markdown.

We can copy the way hugo is handling them: a shortcode folder and we can call the corresponding template by doing `format!("shortcodes/{}.html", shortcode_name).

Most of the bikeshedding is on how the syntax would look like. Current thoughts are:

```markdown
{{ youtube(id="...") }} // single line shortcode

{% blockquote(author="someone") %}
{% end %}
```

If we use the same syntax as Hugo (`{{ < > }}`), it will be parsed as inline html, which might be a bit annoying to deal with.

**Comment by Keats** [comment]
Got the above implemented in #24 

EDIT: now in master
